File: teamapps-ux/src/main/java/org/teamapps/ux/component/iframe/IFrame.java
/*-
 * ========================LICENSE_START=================================
 * TeamApps
 * ---
 * Copyright (C) 2014 - 2022 TeamApps.org
 * ---
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * =========================LICENSE_END==================================
 */
package org.teamapps.ux.component.iframe;

import org.teamapps.dto.DtoEventWrapper;
import org.teamapps.dto.UiComponent;
import org.teamapps.dto.DtoIFrame;
import org.teamapps.ux.component.AbstractComponent;
import org.teamapps.ux.component.CoreComponentLibrary;
import org.teamapps.ux.component.TeamAppsComponent;

@TeamAppsComponent(library = CoreComponentLibrary.class)
public class IFrame extends AbstractComponent {

	private String url;

	public IFrame() {
		this(null);
	}

	public IFrame(String url) {
		this.url = url;
	}

	@Override
	public UiComponent createUiClientObject() {
		DtoIFrame uiIFrame = new DtoIFrame().setUrl(url);
		mapAbstractUiComponentProperties(uiIFrame);
		return uiIFrame;
	}

	@Override
	public void handleUiEvent(DtoEventWrapper event) {
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
		sendCommandIfRendered(() -> new DtoIFrame.SetUrlCommand(url));
	}
}

File: teamapps-client/teamapps-client-core-components/src/main/ts/component/IFrame.ts
/*-
 * ========================LICENSE_START=================================
 * TeamApps
 * ---
 * Copyright (C) 2014 - 2022 TeamApps.org
 * ---
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * =========================LICENSE_END==================================
 */
import {AbstractLegacyComponent, parseHtml, ServerChannel, TeamAppsUiContext} from "teamapps-client-core";
import {DtoIFrame, DtoIFrameCommandHandler} from "../generated";


export class IFrame extends AbstractLegacyComponent<DtoIFrame> implements DtoIFrameCommandHandler {

	private $iframe: HTMLIFrameElement;

	constructor(config: DtoIFrame, serverChannel: ServerChannel) {
		super(config, serverChannel);
		this.$iframe = parseHtml(`<iframe class="IFrame" src="${config.url}"></iframe>`);
		// this.$iframe.addEventListener("load", e => {
		// 	console.log(`load: ${this.$iframe.src}`);
		// });
		// this.$iframe.addEventListener("unload", e => {
		// 	console.log(`unload: ${this.$iframe.src}`);
		// });
	}

	public doGetMainElement(): HTMLElement {
		return this.$iframe;
	}


	setUrl(url: string): void {
		this.$iframe.src = url;
	}

}


